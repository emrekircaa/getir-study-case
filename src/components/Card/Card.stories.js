import { Card } from ".";

const meta = {
  title: "Components/Card",
  component: Card,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
};

export default meta;

export const small = {
  args: {
    size: "sm",
  },
};
export const large = {
  args: {
    size: "lg",
  },
};
