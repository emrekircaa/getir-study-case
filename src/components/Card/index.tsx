import React from "react";
import styled from "styled-components";
import { CardSizeType } from "../../types";
import { getCardWidth } from "../../utils/cardSize";
import { colors } from "../../theme";

interface CardProps {
  title: string; // title of the card
  size: CardSizeType; // size of the card
  children?: React.ReactNode;
}

export const Card: React.FC<CardProps> = ({ children, size, title }) => {
  return (
    <StyledCardContainer>
      <StyledCardTitle>{title}</StyledCardTitle>
      <StyledCard data-testid="card-item-container" size={size}>
        {children}
      </StyledCard>
    </StyledCardContainer>
  );
};

const StyledCardContainer = styled.div``;
const StyledCard = styled.div<{ size: CardSizeType }>`
  box-shadow: 0px 6px 24px rgba(93, 62, 188, 0.04);
  border-radius: 2px;
  padding-top: 24px;
  padding-bottom: 24px;
  width: ${({ size }) => getCardWidth(size)};
  background-color: white;
  gap: 17px;
  display: flex;
  flex-direction: column;
  overflow: hidden;
  input {
    display: block;
    position: relative;
    width: 100%;
  }
  @media (max-width: 1023px) and (min-width: 960px) {
    height: 244px;
  }
  @media (max-width: 959px) {
    height: auto;
    max-width: 576px;
    width: 100%;
  }
`;
const StyledCardTitle = styled.div`
  margin-bottom: 12px;
  font-weight: 600;
  font-size: 13px;
  line-height: 18px;
  color: ${colors.gray500};
`;
