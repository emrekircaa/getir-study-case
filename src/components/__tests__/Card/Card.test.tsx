import { render } from "@testing-library/react";
import { Card } from "../../Card/index";

describe("Card component", () => {
  it("should match snapshot with sm card", () => {
    const { getByTestId } = render(<Card size="sm" title="sample" />);
    const CardContainer = getByTestId("card-item-container");
    expect(CardContainer).toMatchSnapshot();
  });
  it("should match snapshot with lg card", () => {
    const { getByTestId } = render(<Card size="lg" title="sample" />);
    const CardContainer = getByTestId("card-item-container");
    expect(CardContainer).toMatchSnapshot();
  });
  it("should match snapshot with large card", () => {
    const { getByText } = render(
      <Card size="sm" title="sample">
        child
      </Card>
    );
    expect(getByText("child")).toMatchSnapshot();
  });
});
