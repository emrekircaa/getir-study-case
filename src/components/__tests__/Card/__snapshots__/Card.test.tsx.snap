// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot with large card 1`] = `
<div
  class="sc-gEvEer kENWwO"
  data-testid="card-item-container"
>
  child
</div>
`;

exports[`Card component should match snapshot with lg card 1`] = `
<div
  class="sc-gEvEer eAIEtN"
  data-testid="card-item-container"
/>
`;

exports[`Card component should match snapshot with sm card 1`] = `
<div
  class="sc-gEvEer kENWwO"
  data-testid="card-item-container"
/>
`;
